src/main/java/io/stargate/sgv2/jsonapi/exception/playing/DocumentException.java

```diff
@@ -0,0 +1,33 @@
+package io.stargate.sgv2.jsonapi.exception.playing;
+
+/**
+ * Errors related to the document user is trying to insert or update, could be for collections or
+ * tables
+ *
+ * <p>See {@link APIException} for steps to add a new code.
+ */
+public class DocumentException extends RequestException {
+
+  public static final Scope SCOPE = Scope.DOCUMENT;
+
+  public DocumentException(ErrorInstance errorInstance) {
+    super(errorInstance);
+  }
+
+  public enum Code implements ErrorCode<DocumentException> {
+    MISSING_PRIMARY_KEY_COLUMNS,
+    UNKNOWN_TABLE_COLUMNS,
+    UNSUPPORTED_COLUMN_TYPES;
+
+    private final ErrorTemplate<DocumentException> template;
+
+    Code() {
+      template = ErrorTemplate.load(DocumentException.class, FAMILY, SCOPE, name());
```

Comment:
it was intentional to do in the ctor so that the enum will fail when the JVM is compiling the code / first time the Enum is checked if there is an error code that is not in the yaml file. Loading the yaml file happens once and is cached. 

wanted something like that to reduce the risk of failing to create an ApiException during execution of a request, prefer to fail startup

Reply:
Ok just wanted to confirm; makes sense.